pkg/datapath/loader/loader.go

```diff
@@ -732,13 +732,29 @@ func (l *loader) Unload(ep datapath.Endpoint) {
 		}
 	}
 
+	log := log.WithField(logfields.EndpointID, ep.StringID())
+
+	// Remove legacy tc attachments.
+	if err := removeTCFilters(ep.InterfaceName(), netlink.HANDLE_MIN_INGRESS); err != nil {
+		log.WithError(err).Errorf("Removing ingress filter from interface %s", ep.InterfaceName())
+	}
+	if err := removeTCFilters(ep.InterfaceName(), netlink.HANDLE_MIN_EGRESS); err != nil {
+		log.WithError(err).Errorf("Removing egress filter from interface %s", ep.InterfaceName())
+	}
+
```

Comment:
call me crazy. But if we want to be really paranoid about network policy bypass, how about installing two trivial "drop all" programs instead? And letting the kernel dismantle them as it tears down the veth pair.

Reply:
Correct me if I'm wrong, but I think endpoint teardown only takes place after pid1 has died in the container and the container runtime invoked a CNI DEL. A theoretical policy bypass could never elicit a reply other than maybe ICMP from the network stack maintained by the kernel in the namespace (due to an IP being bound there etc.).

We don't need bpf to drop packets using tc, we could install something dumb like `netem loss 100%`, but then we may as well shoot the veth instead. Currently, this is done at the end of cilium-cni's Cmd.Del(). If we really need to address this, we could shift veth removal into the agent as a follow-up, but that could be a lot of work depending on assumptions made in other parts of the chain. Managing pod veths has always been the responsibility of the CNI, and we should probably keep it that way.

Isn't routing teardown enough? That should cut off packets in the endpoint routing case, not sure what the other paths are for pushing a packet into the veth.

Reply:
You can ifdown the veth device, then this is similar as drop-all. For netkit the driver internally does drop all if no BPF is attached if configured as such which is the case for the Cilium work.

Reply:
> You can ifdown the host-facing lxc veth device

You know, sometimes the solution stares back at you. Thanks, I'll give that a try.

Reply:
Done, now setting the interface down after removing routes and before removing tc filters.

Reply:
Nice!